tidy3d/components/data/monitor_data_n2f.py

```diff
@@ -478,40 +572,24 @@ def power(self, medium: Medium = Medium(permittivity=1)) -> xr.Dataset:
             Power at points relative to the local origin.
         """
 
-        frequencies = np.array(self.monitor.freqs)
-
-        eta = np.atleast_1d(self.eta(frequencies, medium))
-        wave_number = np.atleast_1d(self.k(frequencies, medium))
-
-        e_theta = -(self.Lphi.values + eta[None, None, ...] * self.Ntheta.values)
-        e_phi = self.Ltheta.values - eta[None, None, ...] * self.Nphi.values
-
-        values = np.zeros((len(self.monitor.x), len(self.monitor.y), len(frequencies)))
-
-        for i, _x in enumerate(self.monitor.x):
-            for j, _y in enumerate(self.monitor.y):
-                x_glob, y_glob, z_glob = self.monitor.unpop_axis(
-                    self.monitor.plane_distance, (_x, _y), axis=self.monitor.plane_axis
-                )
-                r, _, _ = self.car_2_sph(x_glob, y_glob, z_glob)
-                phase = -1j * wave_number * np.exp(1j * wave_number * r) / (4 * np.pi * r)
-
-                Et = -phase * e_theta[i, j, :]
-                Ep = phase * e_phi[i, j, :]
-
-                Ht = -Ep / eta
-                Hp = Et / eta
-
-                power_theta = 0.5 * np.real(Et * np.conj(Hp))
-                power_phi = 0.5 * np.real(-Ep * np.conj(Ht))
-                values[i, j, :] = power_theta + power_phi
+        if r is not None:
+            log.warning(
+                "'r' supplied to 'Near2FarCartesianData.power' will not be used. "
+                "distance is determined from the coordinates contained in the data."
+            )
 
-        dims = ("x", "y", "f")
-        coords = {"x": np.array(self.monitor.x), "y": np.array(self.monitor.y), "f": frequencies}
+        # get the polar components of the far field
+        fields_sph = self.fields_sph(medium=medium)
```

Comment:
In my analytical dipole test, the Cartesian power is off by a large constant factor, because `self.fields_sph` here doesn't take the `r` distance into account. I think this should be
```python
r_values, _, _ = self.spherical_coords
fields_sph = self.fields_sph(r=r_values, medium=medium)
```
I tested it and it works

Reply:
thanks, changed it to this.